interpolation/src/lib.rs

```diff
@@ -12,22 +12,29 @@ use p3_field::{
     two_adic_coset_zerofier, ExtensionField, Field, TwoAdicField,
 };
 use p3_matrix::MatrixRows;
-use p3_util::log2_strict_usize;
+use p3_util::{log2_strict_usize, reverse_bits_len};
 
 /// Given evaluations of a batch of polynomials over the canonical power-of-two subgroup, evaluate
 /// the polynomials at `point`.
-pub fn interpolate_subgroup<F, EF, Mat>(subgroup_evals: &Mat, point: EF) -> Vec<EF>
+pub fn interpolate_subgroup<F, EF, Mat, const IN_BITREV: bool>(
```

Comment:
Hm since these already work with `MatrixRows`, maybe the callers could pass in one that lazily bit-reverses row indices?

(Although, keeping in mind we'll want to vectorize this at some point; seems like doing that nicely requires a more specific type like `MatrixRowSlices`)

In any case, feel free to leave as is for now.

Reply:
Yeah, it definitely would be cleaner for API consumers to work with transparently-rewritten Matrix objects, but I decided not to for now because of
1. opportunities for hand-optimizing `IN_BITREV` and `!IN_BITREV` paths
2. I suspect we will hit the limits of rust types, stuff like:
- make `mat.bit_rev_view().bit_rev_view()` a no-op
- each Dft implementation may want to return either `BitRevView<T>` or just `T`, and it seems like a lot of boilerplate trait bounds to do that unboxed

IMO we can wait a bit for all the proof systems / API consumers to settle and then try and make things more dependently typed if possible